src/msg/Messenger.cc

```diff
@@ -36,7 +37,7 @@ Messenger *Messenger::create(CephContext *cct, const string &type,
 #ifdef HAVE_XIO
   else if ((type == "xio") &&
 	   cct->check_experimental_feature_enabled("ms-type-xio"))
-    return new XioMessenger(cct, name, lname, nonce, features);
+    return new XioMessenger(cct, name, lname, nonce, features, opts);
```

Comment:
I think we should go ahead and pass `opts` into the other Messengers' constructors while we're at it